### ISSUE TITLE
PHP8.1 support

### ISSUE THREAD

**Issue by Jeckerson** [box-project/box#562]
## Feature Request

Current fatal error that I get with latest version of box:
```
Fatal error: During inheritance of IteratorAggregate: Uncaught ErrorException: Return type of _HumbugBox113887eee2b6\Symfony\Component\Console\Helper\HelperSet::getIterator() should either be compatible with IteratorAggregate::getIterator(): Traversable, or the #[ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in phar:///srv/box.phar/vendor/symfony/console/Helper/HelperSet.php:44
```

Make support for PHP8.1 possible.

**Comment by Jeckerson** [comment]
According Symfony they already support PHP 8.1.

**Comment by Jeckerson** [comment]
Also, this feature really blocks generation of `.phar` which then can be used inside Phalcon tests.

**Comment by theofidry** [comment]
>According Symfony they already support PHP 8.1.

The problem is not Symfony on which the app is built but PHP-Scoper which is used to bundle the PHAR.

Thinking about this again, since Box does not use any PHP 8.0 or 8.1 features, maybe it won't require more than "just" updating the dependencies.

**Comment by Jeckerson** [comment]
So you suggest to compile `.phar` with PHP7.4 and it might work with PHP >= 8.0 ?

**Comment by theofidry** [comment]
No I meant to build box on PHP 8.1

**Comment by Jeckerson** [comment]
@theofidry Do you need any help with this?

**Comment by jrfnl** [comment]
Related #567 / PR #568

**Comment by jrfnl** [comment]
Also related PR #570

**Comment by llaville** [comment]
@theofidry has right when he said that PHP 8.1 compatibility is broken by `humbug/php-scoper` (https://github.com/box-project/box/blob/3.14.0/composer.json#L51) that did not support a compatible `symfony/console` version (see https://github.com/symfony/console/commit/9a4fcecbcccf4b55f4c28e1c71e3cd83d7be99b8 and tags for minimum versions)

Confirmed if you run `composer why symfony/console` on box-project
```
humbug/box          3.x-dev  requires   symfony/console (^4.3.5 || ^5.2)
humbug/php-scoper   0.14.1   requires   symfony/console (^3.2 || ^4.0)
symfony/var-dumper  v5.3.8   conflicts  symfony/console (<4.4)
```

composer.lock told us that it's `symfony/console  v4.4.23` installed in box-project, and minimum should be 4.4.30 (related to https://github.com/symfony/console/commit/9a4fcecbcccf4b55f4c28e1c71e3cd83d7be99b8)

**Comment by andypost** [comment]
Filed another PR to fix requirement checker #585

Somehow latest [release 3.14.1](https://github.com/box-project/box/releases/tag/3.14.1) has no `box.phar` somehow

**Comment by theofidry** [comment]
Should be now working with 3.15
